app.rb

```diff
@@ -16,84 +16,88 @@ def clear_popolo
-                totals[:overall][gender] += 1
-                totals[:overall][totals_seen[p.id]] -= 1
-                totals_seen[p.id] = gender
-            end
-            person_memberships[p.id].each do |m|
-              term_id = m.document[:legislative_period_id]
-              terms[term_id][:overall][gender] += 1
-              terms[term_id][:overall][:total] += 1
-              party = m.on_behalf_of_id
-              terms[term_id][:parties][party][gender] += 1
-              terms[term_id][:parties][party][:total] += 1
-              party_person = party + ':' + p.id
-              if not totals_seen[party_person]
-                  totals[:parties][party][gender] += 1
-                  totals[:parties][party][:total] += 1
-                  totals_seen[party_person] = gender
-              elsif totals_seen[party_person] != gender and gender != 'unknown'
-                  totals[:overall][party][gender] += 1
-                  totals[:overall][party][totals_seen[party_person]] -= 1
-                  totals_seen[party_person] = gender
-              end
+          person_memberships[p.id].each do |m|
+            term_id = m.document[:legislative_period_id]
+            terms[term_id][:overall][gender] += 1
+            terms[term_id][:overall][:total] += 1
+            party = m.on_behalf_of_id
+            terms[term_id][:parties][party][gender] += 1
+            terms[term_id][:parties][party][:total] += 1
+            party_person = party + ':' + p.id
+            if not totals_seen[party_person]
+                totals[:parties][party][gender] += 1
+                totals[:parties][party][:total] += 1
+                totals_seen[party_person] = gender
+            elsif totals_seen[party_person] != gender and gender != 'unknown'
+                totals[:overall][party][gender] += 1
+                totals[:overall][party][totals_seen[party_person]] -= 1
+                totals_seen[party_person] = gender
             end
           end
-
-          legislature[:terms] = terms
-          legislature[:totals] = totals
-          legislatures.push(legislature)
-          l.clear_popolo
         end
-        country[:legislatures] = legislatures
-        gender_stats.push(country)
+
+        legislature[:terms] = terms
+        legislature[:totals] = totals
+        legislatures.push(legislature)
+        l.clear_popolo
       end
+      country[:legislatures] = legislatures
+      gender_stats.push(country)
+    end
+    gender_stats
+  end
+end
 
-      File.write('stats.json', gender_stats.to_json)
+class GBCountryStatsGenerator
+  include Sidekiq::Worker
+  include Everypoliticianbot::Github
+
+  def perform(countries_json_url)
+    Everypolitician.countries_json = countries_json_url
+    with_git_repo('everypolitician/gender-balance-country-stats', branch: 'gh-pages', message: 'Update stats.json') do
+      File.write('stats.json', GenderStats.report.to_json)
```

Comment:
When I first wrote that I was simply thinking of looking at the output as a whole. But of course this output is being committed, so it's actually even more important that the diffs are obvious.